rainier-cats/src/main/scala/com/stripe/rainier/cats/package.scala

```diff
@@ -83,3 +126,39 @@ private[cats] class MonadRandomVariable extends Monad[RandomVariable] {
       case Right(b) => RandomVariable(b)
     }
 }
+
+private[cats] trait EqInstances {
+
+  def eqBigDecimal(epsilon: Double): Eq[BigDecimal] =
+    Eq.instance { (left, right) =>
+      if (right == BigDecimal(0.0))
+        left.abs < epsilon
+      else ((left - right).abs / right.abs) < epsilon
+    }
+
+  def eqReal(epsilon: Double): Eq[Real] = {
+    val bde = eqBigDecimal(epsilon)
+    Eq.instance { (left, right) =>
+      (left, right) match {
+        case (Infinity, Infinity) | (NegInfinity, NegInfinity) => true
+        case (Constant(a), Constant(b))                        => bde.eqv(a, b)
+        case _                                                 => false
```

Comment:
Even without going to a full DAG-equality, what about adding another case for when they are identically the same object?

Reply:
Boom, done. Let me add a TODO as well, as you've suggested.